common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -1268,19 +1273,21 @@ def _update_single_item(self, location, update, allow_not_found=False):
         Set update on the specified item, and raises ItemNotFoundError
         if the location doesn't exist
         """
-        bulk_record = self._get_bulk_ops_record(location.course_key)
-        bulk_record.dirty = True
-        # See http://www.mongodb.org/display/DOCS/Updating for
-        # atomic update syntax
-        result = self.collection.update(
-            {'_id': location.to_deprecated_son()},
-            {'$set': update},
-            multi=False,
-            upsert=allow_not_found,
-            w=1,  # wait until primary commits
-        )
-        if result['n'] == 0:
-            raise ItemNotFoundError(location)
+        course_key = location.course_key
+        with self.bulk_operations(course_key):
```

Comment:
I have tests to ensure the signal is fired once when a course is created, a single item is updated, or an item is published. Please take a look at `test_mixed_modulestore.py` to ensure all cases are covered.